### PR TITLE
Add comprehensive specs for tool execution failure handling

### PR DESCRIPTION
**Pull request by emk-klaay**
- Add specs to ensure conversation continues when tool calls fail
- Test error logging and error message propagation as tool responses
- Verify state remains :in_progress to allow conversation continuation
- Cover different error types: StandardError, ArgumentError, RuntimeError
- Test edge cases: tool not found, missing tool call IDs, multiple failures
- Ensure assistant can continue processing after tool failures

### CODE REVIEW

File: lib/langchain/assistant.rb [hunk 1]
```diff
@@ -322,7 +322,11 @@ def execute_tools
       :in_progress
     rescue => e
       Langchain.logger.error("#{self.class} - Error running tools: #{e.message}; #{e.backtrace.join('\n')}")
-      :failed
+      
+      # Add the error message as a tool response so the AI can see what went wrong
+      add_message(role: @llm_adapter.tool_role, content: "Error: #{e.message}", tool_call_id: messages.last.tool_calls.first&.dig("id"))
```
Comment (alb-klaay) [inline lib/langchain/assistant.rb]: How do we know it's the first tool call that threw an error?

Comment (alb-klaay) [reply lib/langchain/assistant.rb]: Maybe we should move the rescue to the `run_tool` method, since we have required info there?

https://github.com/klaayinc/langchainrb/blob/f7de8f6b62a2c525817cac93a702a3b732173e63/lib/langchain/assistant.rb#L372

Comment (emk-klaay) [reply lib/langchain/assistant.rb]: Yeah, I also thought about that.

**Review by alb-klaay — APPROVED** [review]

File: lib/langchain/assistant.rb [hunk 2]
```diff
@@ -374,6 +371,17 @@ def run_tool(tool_call)
       else
         submit_tool_output(tool_call_id: tool_call_id, output: output)
       end
+    rescue => e
+      Langchain.logger.error("#{self.class} - Error running tool '#{tool_name}': #{e.message}; #{e.backtrace.join('\n')}")
+      
+      # Add the error message as a tool response so the AI can see what went wrong
+      # Handle both ToolResponse and legacy return values for error cases
+      content = "Error running tool: #{e.message}"
```
Comment (emk-klaay) [inline lib/langchain/assistant.rb]:
```suggestion
      content = "Error running tool (#{tool_name}): #{e.message}"
```
